src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -219,6 +225,13 @@ public long getMaxMemoryLimitInBytes() {
     return (val instanceof String) ? Long.parseLong(val.toString()) : (long) val;
   }
 
+  /** @return Return whether memory optimization for Parquet is enabled. */
+  public boolean getEnableParquetMemoryOptimization() {
```

Comment:
I think you also need to add it to `setParameterMap()` otherwise `setParameterOverrides()` in the client will not work, e.g. from json profile, like we do for `BLOB_FORMAT_VERSION_DEFAULT`.